### PR TITLE
Update to recent versions + make handling server / client more symmetric

### PR DESCRIPTION
**Pull request by mindreframer**
Hey @LostKobrakai , because I wanted to use depot with depot_s3, I kinda had to fix / update minio_server. 

Not sure if you are still using this library, it kinda looked abandoned to me. I took the liberty to move logic around to make it more regular and simpler. Currently there is a lot of simple code duplication, yet I did not want to introduce more abstractions and also wanted to keep the cognitive overhead somewhat lower. 

Feel free to close this PR, is you are not using minio_server anymore or maybe reuse some ideas. 

Cheers and have a good day!

### CODE REVIEW

**Comment by LostKobrakai** [comment]
Hey @mindreframer. Thanks for all that work. While I haven't had time to dedicate to this in a while I'm very much open to improvements. From a quick glance this looks like a lot of great improvements to the codebase. I do want to take a more thorough look at it some time this week and then merge the changes.

Also great to seem some usage for depot/_s3. The s3 adapter is not fully up to date with the API newest api of depot, but should still be usable.

File: mix.exs [hunk 1]
```diff
@@ -38,7 +38,7 @@ defmodule MinioServer.MixProject do
       {:ex_aws_s3, "~> 2.0", only: [:dev, :test]},
       {:hackney, "~> 1.15", only: [:dev, :test]},
       {:sweet_xml, "~> 0.6", only: [:dev, :test]},
-      {:ex_doc, "~> 0.21", only: :dev, runtime: false}
+      {:ex_doc, "~> 0.25", only: :dev, runtime: false}
     ]
   end
```
Comment (LostKobrakai) [inline mix.exs:44]: The package callback needs to be updated to work with the new version json files

File: README.md [hunk 2]
```diff
@@ -31,9 +31,23 @@ The minio binary is not included in the package to save on space. But you can
...
 ```sh
+# with menu to select arch / version
 mix minio_server.download
-# or directly
+
+# download the latest SERVER binary for darwin-amd64
 mix minio_server.download --arch darwin-amd64 --version latest
+
+# download the latest CLIENT binary for darwin-amd64
+mix minio_server.download --client --arch darwin-amd64 --version latest
```
Comment (LostKobrakai) [inline README.md:41]: Is there a reason you split downloading the server and client apart for the mix task? I'd expect people to generally want to have both.

File: lib/minio/config.ex [hunk 3]
```diff
@@ -0,0 +1,81 @@
+defmodule MinioServer.Config do
+  alias MinioServer.CpuInfo
+  @doc "A list of all the available architectures downloadable."
+  @spec available_architectures :: [MinioServer.architecture()]
+  def available_architectures do
+    ["windows-amd64", "darwin-amd64", "darwin-arm64", "linux-amd64", "linux-arm", "linux-arm64"]
```
Comment (LostKobrakai) [inline lib/minio/config.ex]: Maybe put this in multiple lines for better glancability.

File: lib/minio/cpu_info.ex [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  defp cpu_type_sub(os_type) when os_type in [:freebsd, :macos] do
+    confirm_executable("uname")
+    case System.cmd("uname", ["-m"]) do
+      {result, 0} -> result |> String.trim()
+      _ -> raise RuntimeError, message: "uname does not work."
+    end
+  end
+
+  defp confirm_executable(command) do
+    if is_nil(System.find_executable(command)) do
```
Comment (LostKobrakai) [inline lib/minio/cpu_info.ex]: This could be `unless System.find_executable(command) do`

**Review by LostKobrakai — COMMENTED** [review]
Thanks again for the work. I've had a chance to look at this today. I have a few questions/minor suggestions, though I'm really excited about the changes as a whole.

File: mix.exs (hunk 1 above)
Comment (mindreframer) [reply mix.exs:44]: @LostKobrakai Not sure I follow. The `package` callback looks like this one `https://github.com/jonasschmidt/ex_json_schema/blob/master/mix.exs#L55-L62` and seems to work....

File: README.md (hunk 2 above)
Comment (mindreframer) [reply README.md:41]: The reason was to simplify code and also handling of possible error cases. There is little cost to calling a command twice with different arguments, but the code complexity makes handling all edge cases somewhat harder.

File: mix.exs (hunk 1 above)
Comment (LostKobrakai) [reply mix.exs:44]: This is about line 48. For whatever reason I couldn't put a comment on that line.

Comment (mindreframer) [reply mix.exs:44]: `      files: ~w(lib .formatter.exs mix.exs versions.json README* LICENSE*),` ? 
What should be changed about this line?

Comment (LostKobrakai) [reply mix.exs:44]: The `versions.json` you renamed / split.

File: README.md (hunk 2 above)
Comment (LostKobrakai) [reply README.md:41]: Ok, sounds fair.

File: mix.exs (hunk 1 above)
Comment (mindreframer) [reply mix.exs:44]: duh... now I got this. :)

**Comment by mindreframer** [comment]
@LostKobrakai thx!

**Pull request merged**